### PR TITLE
Java testing and debugging docs page update

### PR DESCRIPTION
**Pull request by tsurdilo**
Signed-off-by: Tihomir Surdilovic <tihomir@temporal.io>

## What does this PR do?
Updates the testing and debugging java page:
1) adds dependency info
2) small text updates
3) section labels

### CODE REVIEW

**Comment by tsurdilo** [comment]
so prettier check is failing because i am cutting the prs from my fork of temporalio/documentation :) 
If you want me to I can redo this pr, if it's ok to merge this one I will from now on work on origin so the check should work fine.

File: docs/java/testing-and-debugging.md [hunk 1]
```diff
@@ -12,6 +14,31 @@ easily test long-running Workflows in seconds, without having to change your Wor
 You can use the provided `TestWorkflowEnvironment` with a Java unit testing framework of your choice,
 such as JUnit.
 
+## Setup testing dependency
+In order to start using the Java SDK test framework you need to add [`io.temporal:temporal-testing`](https://search.maven.org/artifact/io.temporal/temporal-testing)
```
Comment (djmagee) [inline docs/java/testing-and-debugging.md:18]: Recommend changing "In order to" to simply "To", per the "in order to" entry in [Word List](https://developers.google.com/style/word-list#letter-i) in the Google dev doc style guide.

Comment (djmagee) [inline docs/java/testing-and-debugging.md:18]: Suggest inserting a comma in "framework you" for readability.

File: docs/java/testing-and-debugging.md [hunk 2]
```diff
@@ -12,6 +14,31 @@ easily test long-running Workflows in seconds, without having to change your Wor
...
+</dependency>
+```
+
+**[Gradle Groovy DSL](https://gradle.org/)**:
+
+```groovy
+testImplementation group: 'io.temporal', name: 'temporal-testing', version: '1.0.7'
+```
+
+Make sure to set the version which matches your dependency version of the [Temporal Java SDK](https://github.com/temporalio/sdk-java).
```
Comment (djmagee) [inline docs/java/testing-and-debugging.md:38]: Change "which" to "that" to make the phrase restrictive.

File: docs/java/testing-and-debugging.md [hunk 3]
```diff
@@ -110,6 +137,10 @@ public class HelloActivityTest {
 }
 ```
 
+You can find all unit tests for the [Temporal Java samples](https://github.com/temporalio/samples-java) repository [here](https://github.com/temporalio/samples-java/tree/master/src/test/java/io/temporal/samples).
```
Comment (djmagee) [inline docs/java/testing-and-debugging.md:140]: For SEO, recommend changing link text to something like "repository in [temporal/samples-java]", per [Link text](https://developers.google.com/style/link-text) in the Google dev doc style guide. (Finding suitable link text when linking into a repo is definitely trickier than  when linking to an ordinary page.)

**Comment by tsurdilo** [comment]
@djmagee thanks for the review! Updated

File: docs/java/testing-and-debugging.md (hunk 1 above)
Comment (djmagee) [inline docs/java/testing-and-debugging.md]: - Recommend changing "In order to" to simply "To", per the "in order to" entry in [Word List](https://developers.google.com/style/word-list#letter-i) in the Google dev doc style guide.
- Suggest inserting a comma in "framework you" for readability.

```suggestion
To start using the Java SDK test framework, you need to add [`io.temporal:temporal-testing`](https://search.maven.org/artifact/io.temporal/temporal-testing) 
```

File: docs/java/testing-and-debugging.md (hunk 2 above)
Comment (djmagee) [inline docs/java/testing-and-debugging.md]: Change "which" to "that" to make the phrase restrictive.

```suggestion
Make sure to set the version that matches your dependency version of the [Temporal Java SDK](https://github.com/temporalio/sdk-java).
```

File: docs/java/testing-and-debugging.md (hunk 3 above)
Comment (djmagee) [inline docs/java/testing-and-debugging.md]: For SEO, recommend changing link text to something like "repository in [temporalio/samples-java]", per [Link text ](https://developers.google.com/style/link-text)in the Google dev doc style guide. (Finding suitable link text when linking into a repo is definitely trickier than when linking to an ordinary page.)

```suggestion
You can find all unit tests for the [Temporal Java samples](https://github.com/temporalio/samples-java) repository in [temporalio/samples-java](https://github.com/temporalio/samples-java/tree/master/src/test/java/io/temporal/samples).
```

**Review by djmagee — APPROVED** [review]

**Comment by tsurdilo** [comment]
@flossypurse can we merge or do we need to make more changes/updates?

**Review by flossypurse — APPROVED** [review]

**Pull request merged**